### PR TITLE
fix: board access and other fixes

### LINKED ISSUE #810: [BUG] Alpha numeric texts in sign up form
## Description
As a user I should be able to sign up using my personal details

### Current behavior
Alpha numeric characters aren't being allowed on the sign up form

Image

### LINKED ISSUE #813: [BUG] Create team button
## Description
As a user I should be able to create a team

### Current behavior
The "Create team" button is spammable, meaning it's possible to create multiple teams in quick succession

Image

### LINKED ISSUE #815: [BUG]: Board access
## Description

A user that isn't participating in a board is allowed to see the board's page.

### LINKED ISSUE #816: [BUG] Card unmerge
## Description
As a user I should be able to unmerge cards, given that I have the permissions to do so

### Current behavior
Any user is allowed to unmerge hidden cards, even tho they aren't able to see the 3 dots menu

Image

### LINKED ISSUE #817: [BUG]: board settings
## Description

The switch buttons in the board settings don't change their state when a board is updated board update changes

### PR DESCRIPTION
**Pull request by nunocaseiro**
Fixes #816
Fixes #815 
Fixes #813 
Fixes #810 
Fixes #817

## Proposed Changes

  - Protect a board from a user that isn't participating or doesn't have privileges 
  - Disable create team after first successful submit 
  - Remove alphanumeric rule in register fields 
  - Avoid unmerge when the cards are hidden
  - Fix board settings switches

This pull request closes #816 and closes #815 and closes #813 and closes #810 and closes #817

### CODE REVIEW

**Comment by miguel-felix1** [comment]
Can still see boards even tho my user is not in a team
Screenshot 2023-01-06 at 14 10 02

**Comment by nunocaseiro** [comment]
> Can still see boards even tho my user is not in a team
> Screenshot 2023-01-06 at 14 10 02

Thanks for the feedback we should address this in another issue. @patricia-mdias

File: frontend/src/components/Board/Settings/index.tsx [hunk 1]
```diff
@@ -101,32 +109,45 @@ const BoardSettings = ({
    * if yes set the input with this value
    */
   useEffect(() => {
-    if (!isOpen && board) {
+    if (board) {
```
Comment (rpvsilva) [inline frontend/src/components/Board/Settings/index.tsx:112]: You can destructure `board` variable

**Review by miguel-felix1 — APPROVED** [review]
LGTM 🚀

**Review by rpvsilva — CHANGES_REQUESTED** [review]

File: frontend/src/components/Board/Settings/index.tsx [hunk 2]
```diff
@@ -109,24 +109,22 @@ const BoardSettings = ({
...
-      setSwitchesState((prev) => ({
-        ...prev,
-        hideVotes: board.hideVotes,
-        hideCards: board.hideCards,
-        maxVotes: !isEmpty(board.maxVotes),
-      }));
-    }
+    setData((prev) => ({
+      ...prev,
+      title: board.title,
```
Comment (rpvsilva) [inline frontend/src/components/Board/Settings/index.tsx:114]: you can do a object destructuring before using board properties

```js
const { title, maxVotes, ...... } = board;
```

**Review by rpvsilva — APPROVED** [review]

**Review by rpvsilva — APPROVED** [review]

**Pull request merged**